api/v1/overlaynetwork_types.go

```diff
@@ -20,34 +20,47 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // OverlayNetworkSpec defines the desired state of OverlayNetwork
 type OverlayNetworkSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Name is the NetworkAttachmentDefinition name.
+	Name string `json:"name"`
 
-	// Foo is an example field of OverlayNetwork. Edit overlaynetwork_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
-}
+	// NetworkName is the name of the overlay network.
+	// The following name formats are supported:
+	// - <namespace>/<name> : Used when cross namespace connectivity is desired.
+	// - <name> : The name will be prefixed with the current namespace to assure uniqueness across namespaces.
+	NetworkName string `json:"networkName"`
+
+	// The maximum transmission unit (MTU).
+	// The default value, 1300, is automatically set by the kernel.
+	// +optional
+	Mtu string `json:"mtu,omitempty"`
+
+	// The subnet to use for the network across the cluster.
+	// Only include the CIDR for the node. E.g. 10.100.200.0/24.
+	//
+	// IPv6 (2001:DBB::/64) and dual-stack (192.168.100.0/24,2001:DBB::/64) subnets are supported.
+	//
+	// When omitted, the logical switch implementing the network only provides layer 2 communication,
+	// and users must configure IP addresses for the pods.
+	// Port security only prevents MAC spoofing.
+	// +optional
+	Subnets string `json:"subnets,omitempty"`
 
-// OverlayNetworkStatus defines the observed state of OverlayNetwork
-type OverlayNetworkStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// A comma-separated list of CIDRs and IP addresses.
+	// IP addresses are removed from the assignable IP address pool and are never passed to the pods.
+	// +optional
+	excludeSubnets string `json:"excludeSubnets,omitempty"`
 }
 
 //+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
 
 // OverlayNetwork is the Schema for the overlaynetworks API
 type OverlayNetwork struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   OverlayNetworkSpec   `json:"spec,omitempty"`
-	Status OverlayNetworkStatus `json:"status,omitempty"`
```

Comment:
The conventions is having both `spec` and `status`, why Status is removed?

Reply:
@AlonaKaplan [suggested](https://github.com/AlonaKaplan/selfserviceoverlay/pull/10#discussion_r1514640924) to remove it if we are not using it.

Reply:
I think its better to leave `status`, the most common convention is having them both [[1]](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status)
It also scale better as we wont need to introduce it again when we want to provide some feedback in regard to the CRD instance creation.

[1] https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

Reply:
We are currently not reflecting the current state, therefore I agreed with Alona to remove it.
If we will reflect about the current state in the reconciliation (we should!), it will be required.

In the ref you provided, I do not think it is a must to have it.

I do not have a hard opinion on this matter, so whatever @AlonaKaplan favors I will change to comply.

Reply:
I don't mind, your decision:)